src/plugins/charts/public/static/components/warnings.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiButtonEmpty, EuiIcon, EuiNotificationBadge, EuiPopover } from '@elastic/eui';
+import React, { useState } from 'react';
+
+export function Warnings({ warnings }: { warnings: React.ReactNode[] }) {
+  const [open, setOpen] = useState(false);
+  if (warnings.length === 0) return null;
+  return (
+    <>
+      <EuiPopover
+        isOpen={open}
+        closePopover={() => setOpen(false)}
+        button={
+          <EuiButtonEmpty onClick={() => setOpen(!open)}>
+            <EuiNotificationBadge>{warnings.length}</EuiNotificationBadge>{' '}
+            <EuiIcon color="warning" type="alert" />
+          </EuiButtonEmpty>
```

Comment:
Why not use the available `EuiButtonEmpty` props as suggested below, rather than nesting `EuiNotificationBadge` and `EuiIcon`? It would be more consistent with what we're doing in Lens and probably look a bit better. Note that you'll still need to account for i18n and pluralization for the `warnings` text.

```suggestion
          <EuiButtonEmpty color="warning" iconType="alert" onClick={() => setOpen(!open)} size="xs">{warnings.length} warnings</EuiButtonEmpty>
```